internal/runner/operation_status_report.go

```diff
@@ -40,9 +47,71 @@ func (r *Runner) executeStatusReportOp(
 		return nil, err
 	}
 
+	if statusReportResult != nil {
+		err = r.enableApplicationPoll(ctx, log, job.Application)
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return &pb.Job_Result{
 		StatusReport: &pb.Job_StatusReportResult{
 			StatusReport: statusReportResult,
 		},
 	}, nil
 }
+
+// enableApplicationPoll will switch on its queued poll handler. This means
+// on the defined interval, Waypoint will generate a status report for this
+// application. When an application is initially inserted, like on a `waypoint init`,
+// it won't enable the poller to generate a status report. This method switches
+// it on after the first status report is generated. Each time after it should
+// do nothinng when its polling has been enabled.
+func (r *Runner) enableApplicationPoll(
```

Comment:
This feels a little weird to me. My immediate question is: why can't polling be enabled by default, and the poll query just ignores any applications that never had any status reports? This should be solvable in the data layer, and not through a state machine. 

i.e. if somehow all the deployments get deleted, won't the polling still be enabled? That feels somewhat broken. If we solve it in the data layer: only poll things that (1) have a deployment and (2) have at least one status report then it will be resilient to any state changes, right?

Reply:
Ok, this makes more sense to me! I've removed this func in favor of turning on app status polling by default for projects that enable polling and have a remote data source! If the app doesn't have a status report or a deployment, a job for generating a status report won't be queued already by the poll handler.